modules/library/main/src/test/java/org/geotools/data/collection/ListFeatureCollectionTest.java

```diff
@@ -131,6 +141,65 @@ public void iterator() {
         assertTrue(copy.isEmpty());
     }
 
+    @Test
+    public void removeAndAddFeatureBounds() {
+        // create test points
+        createPointFeatures(WORLD, 3);
+        // remove last feature in collection
+        List<SimpleFeature> copy = new ArrayList<SimpleFeature>(featureList);
+        SimpleFeature f = copy.get(2);
+        featureCollection.remove(f);
+        // get the new bounds (removed feature)
+        ReferencedEnvelope postRemoveFeatureBounds = featureCollection.getBounds();
+        // add new feature
+        SimpleFeature newFeature = createPointFeature(10, 4);
+        featureCollection.add(newFeature);
+        // get new bounds
+        ReferencedEnvelope newFeatureBounds = featureCollection.getBounds();
+        // compare new bounds with old....
+        boolean isContained = newFeatureBounds.contains((Envelope) postRemoveFeatureBounds);
+        assertTrue(isContained);
+        // ensure new point bounds don't equal over-all bounds
+        assertNotEquals(newFeatureBounds, newFeature.getBounds());
+    }
+
+    @Test
+    public void threeDimensionalFeatureBounds() {
+        // create FilterFactory for bounding box
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+        // create GeometryFactory for creating a point
+        GeometryFactory gf = new GeometryFactory(new PrecisionModel());
+        SimpleFeatureType type = null;
+        try {
+            type = DataUtilities.createType("testSchema", "name:String,*geom:Geometry");
+        } catch (SchemaException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+
+        // create test features
+        SimpleFeature f1 = SimpleFeatureBuilder.build(type, new Object[] { "testFeature1", gf.createPoint(new Coordinate(10, 20, 30)) }, null);
+        SimpleFeature f2 = SimpleFeatureBuilder.build(type, new Object[] { "testFeature2", gf.createPoint(new Coordinate(10, 10, 60)) }, null);
+        SimpleFeature f3 = SimpleFeatureBuilder.build(type, new Object[] { "testFeature2", gf.createPoint(new Coordinate(1, 10, 6)) }, null);
+
+        // create BBox
+        BoundingBox3D envelope1 = new ReferencedEnvelope3D( 0, 50, 0, 50, 0, 70, null);
+        Filter bbox1 = ff.bbox(ff.property("geom"), envelope1);
+
+        // initialize FC
+        featureCollection = new ListFeatureCollection(TYPE, featureList);
+
+        // add created features to the collection
+        featureCollection.add(f1);
+        featureCollection.add(f2);
+        featureCollection.add(f3);
```

Comment:
I think we want to check the featureCollection.getBounds() method to ensure it is a ReferencedEnvelope3D?